docs/src/dev/future/proposal-multi-threaded-transaction-5.asciidoc

```diff
@@ -0,0 +1,146 @@
+////
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+////
+image::apache-tinkerpop-logo.png[width=500,link="https://tinkerpop.apache.org"]
+
+*x.y.z - Proposal 4*
+
+Proposal for adding support for tx which are not thread bound.
+
+=== Introduction
+The current implementation of transactions and sessions uses approach with thread local variables that requires complex implementation and LifeCycle management. Using non-threaded sessions can improve performance in some scenarios.
+
+=== Motivation
+* Simlify GremlinExecutor. Complex LifeCycle control is now used.
+* Easier integration with HTTP because transactions can be used without being tied to a connection.
+
+=== Assumptions
+
+* Need to reuse existing code as much as possible.
+* Users should be able to work both types of transactions, thread-bound and new non-thread bound. Perhaps after performance testing only one type will be left.
```

Comment:
I think it makes more sense to just phase out the old transaction model. There should ultimately only be one model.

Reply:
Is this statement specifically about the transaction model in TinkerGraph or for all providers who support transactions? I agree that TinkerGraph should eventually converge on a single model, however I imagine there would be some providers who would like to retain a thread-bound model.

Reply:
Will this new model fully replace the current model once it's out in a major release? Or will it be more of deprecating current model, then removing it a major release after? Or are you proposing to keep both indefinitely?